tests/unit/test_gas_strategies.py

```diff
@@ -33,3 +39,36 @@ def test_fixed_price_gas_strategy():
     assert 12340000000 == strategy("web3", "tx")
     assert 12340000000 == strategy("web3", "tx")
     assert "12gwei" == strategy.name
+
+
+def test_max_price_gas_strategy(mocker, monkeypatch):
+
+    gas_prices_gwei = [10, 100, 999, 1000, 1001, 1_000_000, 1_000_000_000]
+    gas_prices_wei = [Web3.toWei(gwei_price, 'gwei') for gwei_price in gas_prices_gwei]
+    max_gas_price_gwei = 1000
+    max_gas_price_wei = Web3.toWei(max_gas_price_gwei, 'gwei')
+    mock_gas_strategy = mocker.Mock(side_effect=itertools.cycle(gas_prices_wei))
+
+    # Let's start assuming there's no limit
+    monkeypatch.delenv(NUCYPHER_ENVVAR_MAX_GAS_PRICE_GWEI, raising=False)
+    wrapped_strategy = max_price_gas_strategy_wrapper(mock_gas_strategy)
+
+    for price in gas_prices_wei:
+        assert wrapped_strategy("web3", "tx") == price
+
+    # If the corresponding envvar has a valid price in gwei, use this as the max gas price
+    monkeypatch.setenv(NUCYPHER_ENVVAR_MAX_GAS_PRICE_GWEI, max_gas_price_gwei)
```

Comment:
Perhaps we can accept it for this PR for the sake of a near-term, but this option seems much more like a character configuration value than an environment variable - At minimum I suggest adding a TODO tied to a follow-up Issue for future follow-up.

```
nucypher ursula config --max-gas-price 150 
```

Reply:
Opened #2485